include/tracelogging/TraceLoggingProvider.h

```diff
@@ -1442,47 +1440,50 @@ If provided, the tags parameter must be an integer value.
         pOut = (char *)memcpy(pOut, pchEventName, cchEventName);
         pOut += cchEventName;
 
-        if (keyword != 0)
+        // ProviderName + ":" + EventName + KeywordSuffix must be less than 256 characters.
+        if (pOutEnd - pOut < 3)
         {
-            // ProviderName + ":" + EventName + KeywordSuffix must be less than 256 characters.
-            if (pOut == pOutEnd)
-            {
-                // If not, truncate KeywordSuffix.
-                _tlg_ASSERT(!"ProviderName+EventName too long");
-            }
-            else
+            // If not, truncate KeywordSuffix.
+            _tlg_ASSERT(!"ProviderName+EventName too long");

```

Comment:
The assert could be clearer here -- ProviderName+EventName+KeywordSuffix too long

Reply:
Same below where adding the number